py3status/modules/arch_updates.py

```diff
@@ -69,21 +69,25 @@ def post_config_hook(self):
             self.include_aur = False
 
     def check_updates(self):
-        pacman_updates = self._check_pacman_updates()
-        aur_updates = self._check_aur_updates()
-        if aur_updates == '?':
-            total = pacman_updates
-        else:
-            total = pacman_updates + aur_updates
+        pacman_updates = aur_updates = total = 0
+
+        if self.include_pacman:
+            pacman_updates = self._check_pacman_updates()
+
+        if self.include_aur:
+            aur_updates = self._check_aur_updates()
+
+        total += pacman_updates if pacman_updates is not None else 0
+        total += aur_updates if aur_updates is not None else 0
 
         if self.hide_if_zero and total == 0:
             full_text = ''
         else:
             full_text = self.py3.safe_format(
                 self.format,
                 {
-                    'aur': aur_updates,
-                    'pacman': pacman_updates,
+                    'aur': aur_updates if aur_updates is not None else '?',
+                    'pacman': pacman_updates if pacman_updates is not None else '?',
                     'total': total,
```

Comment:
I was trying to stay consistent with the original code in showing `?` when `aur_updates` were unknown. Are you suggesting I add a customization to not show when offline? I can do that.

Reply:
>Are you suggesting I add a customization to not show when offline? I can do that.

We can already do this in the `format`. 

Return `pacman_updates` and `aur_updates` as-is (without ifs) to receive `None` instead of an exception when we're offline. Now in the format, if we really want `?`, I think the format would be something like `format = 'UPD: [{pacman}|\?show ?]/[{aur}|\?show ?]'`.